server/src/main/java/org/elasticsearch/common/util/concurrent/ThreadContext.java

```diff
@@ -510,6 +544,13 @@ public <T> T getTransient(String key) {
         return (T) threadLocal.get().transientHeaders.get(key);
     }
 
+    /**
+     * Returns unmodifiable copy of all transient headers.
+     */
+    public Map<String, Object> getTransientHeaders() {
+        return Collections.unmodifiableMap(threadLocal.get().transientHeaders);
+    }
```

Comment:
nit: looks like you don't need this anymore.

Reply:
I kind of do, as it helps me with testing the context stashing. I can make it package protected?